### PR TITLE
add domain representations

### PR DESCRIPTION
**Pull request by steve-chavez**
For https://github.com/PostgREST/postgrest/pull/2839

[Rendered doc](https://postgrest--655.org.readthedocs.build/en/655/references/api/domain_representations.html).

### CODE REVIEW

File: docs/references/api/domain_representations.rst [hunk 1]
```diff
@@ -0,0 +1,181 @@
+.. _domain_reps:
+
+Domain Representations
+######################
+
+Domain Representations separates "how the data is presented" from "how the data is stored". It works by creating `domains <https://www.postgresql.org/docs/current/sql-createdomain.html>`_ and `casts <https://www.postgresql.org/docs/current/sql-createcast.html>`_, the latter act on the former to present the data in different formats.
```
Comment (aljungberg) [inline docs/references/api/domain_representations.rst:6]:
```suggestion
Domain Representations separates "how the data is presented" from "how the data is stored". It works by creating `domains <https://www.postgresql.org/docs/current/sql-createdomain.html>`_ and `casts <https://www.postgresql.org/docs/current/sql-createcast.html>`_, the latter act on the former to present and receive the data in different formats.
```

File: docs/references/api/domain_representations.rst [hunk 2]
```diff
@@ -0,0 +1,181 @@
+.. _domain_reps:
+
+Domain Representations
+######################
+
+Domain Representations separates "how the data is presented" from "how the data is stored". It works by creating `domains <https://www.postgresql.org/docs/current/sql-createdomain.html>`_ and `casts <https://www.postgresql.org/docs/current/sql-createcast.html>`_, the latter act on the former to present and receive the data in different formats.
```
Comment (aljungberg) [inline docs/references/api/domain_representations.rst:6]: Nicely done! I struggled with coming up with a concise way to describe the feature.

**Review by aljungberg — CHANGES_REQUESTED** [review]
We might want a note to remind the dev to reload the schema cache after adding new domains or casts?

File: docs/references/api/domain_representations.rst (hunk 2 above)
Comment (steve-chavez) [reply docs/references/api/domain_representations.rst:6]: Thanks! Actually I stole some phrases from your [opening issue](https://github.com/PostgREST/postgrest/issues/2310#issue-1267408755) :smiley: 

I wanted to remove the "latter/former" words but I found it hard. Although it seems ok leaving it this way.

I've just addressed your latest reviews. LMK if you approve and I'll merge :rocket:.

Comment (steve-chavez) [reply docs/references/api/domain_representations.rst:6]: Also clarified the advantages (focusing on views) and "why not create a type?".

**Review by aljungberg — APPROVED** [review]
Looks good!

**Pull request merged**